presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -138,6 +141,7 @@
     private final TransactionManager transactionManager;
 
     private final ConcurrentMap<String, Collection<ConnectorMetadata>> catalogsByQueryId = new ConcurrentHashMap<>();
+    private final Set<QueryId> queriesWithRegisteredCallbacks = new HashSet<>();
```

Comment:
Do we actually need this?

Reply:
This is to make sure that we add only 1 callback for a query.